### ISSUE TITLE
(core, testing) test groups fail python 3.6

### ISSUE THREAD

**Issue by jmwoloso** [ipython/ipython#10409]
Hello,

Was wondering if someone might have any clues how to correct this. The issue seems to be with the `enum` package and I'm not sure what to do at this point. I've attached the output from running `iptest`. Thanks for your help in advance!

[ipython_test.txt](https://github.com/ipython/ipython/files/869188/ipython_test.txt)

**Comment by Carreau** [comment]
hi @jmwoloso, 

from the errors it seem like your python 3.6 install is broken:
```
/tmp/tmpxr0yzrav/tmpaircv4zt/test.py' produced error: ''import warnings' failed; traceback:
Traceback (most recent call last):
  File "/home/jwolosonovich/.virtualenvs/avaopt/lib/python3.6/warnings.py", line 505, in <module>
    _processoptions(sys.warnoptions)
  File "/home/jwolosonovich/.virtualenvs/avaopt/lib/python3.6/warnings.py", line 186, in _processoptions
    _setoption(arg)
  File "/home/jwolosonovich/.virtualenvs/avaopt/lib/python3.6/warnings.py", line 192, in _setoption
    import re
  File "/home/jwolosonovich/.virtualenvs/avaopt/lib/python3.6/re.py", line 122, in <module>
    import enum
ModuleNotFoundError: No module named 'enum'
```

The error comes from the `re` module, which is from the standard library. 
 In a pure python shell, can you try `>>> import re` ?

**Comment by jmwoloso** [comment]
From the terminal this code works. 
```
python -c 'import re;print("This worked!")'
This worked!
```

**Comment by jmwoloso** [comment]
from the shell, I am able to import `re` as well, of course.

**Comment by Carreau** [comment]
That is a weird bug. We do have CI on 3.6 which version of IPython?

**Comment by jmwoloso** [comment]
```
$ python -c 'import IPython;print(IPython.__version__)'
5.3.0
```

**Comment by Carreau** [comment]
Hum, ok, seem like 5.x was not tested on 3.6, #10411 should fix that (and test are passing on 3.6). Master does pass on 3.6 and the only related things I can see is #10316 / #10232 but that only affect autoreload.

The only other thing I can think of is `https://github.com/pyenv/pyenv/issues/711` and `https://github.com/pypa/virtualenv/issues/962` if you are using pyenv/tox/virtualenv with older versions. Could that be the case?

**Comment by jmwoloso** [comment]
I'll try with master first. The virtualenv-related stuff could be a factor as well (nice find!). 

Initially (a year or two ago) I installed virtualenvenv with python 3.4.3 and I've just been passing the `--python` flag when I create new environments with versions other than 3.4.3.

Thanks for your help! I'll try master first then I'll reinstall virtualenv using 3.6 as the default if master doesn't do the trick.

⁣ ​

**Comment by jmwoloso** [comment]
No dice with Master (see attached).
[ipython_test2.txt](https://github.com/ipython/ipython/files/873106/ipython_test2.txt)

**Comment by minrk** [comment]
@jmwoloso there should be a file `/home/jwolosonovich/.virtualenvs/avaopt/lib/python3.6/enum.py` in your env. If there isn't, the env is broken (it's missing part of Python itself), and you probably need to destroy and recreate it. You might make sure that the env is created with Python 3.6 (e.g. `python3.6 -m venv ...`) if deleting and recreating doesn't work the first time.

**Comment by jmwoloso** [comment]
@Carreau @minrk Thank you both for your help. I upgraded to `virtualenv==15.1.0` and now the tests all pass. It turns out that you were correct, the `enum.py` file was missing and creating new environments with the older version of `virtualenv` failed to pull in everything it should have so I upgraded and everything works.
[ipython_test3.txt](https://github.com/ipython/ipython/files/876889/ipython_test3.txt)
